hazelcast/src/main/java/com/hazelcast/map/impl/KeyValueConsumingEntryProcessor.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright (c) 2008-2020, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.map.impl;
+
+import com.hazelcast.map.EntryProcessor;
+import com.hazelcast.nio.ObjectDataInput;
+import com.hazelcast.nio.ObjectDataOutput;
+import com.hazelcast.nio.serialization.IdentifiedDataSerializable;
+
+import java.io.IOException;
+import java.util.Map;
+import java.util.function.BiConsumer;
+
+public class KeyValueConsumingEntryProcessor<K, V> implements EntryProcessor<K, V, V>, IdentifiedDataSerializable {
+
+    BiConsumer<? super K, ? super V> action;
+
+    public KeyValueConsumingEntryProcessor() {
+    }
+
+    public KeyValueConsumingEntryProcessor(BiConsumer<? super K, ? super V> action) {
+        this.action = action;
+    }
+
+    @Override
+    public V process(Map.Entry<K, V> entry) {
+        action.accept(entry.getKey(), entry.getValue());
+        return entry.getValue();
```

Comment:
Since `forEach` doesn't use these values, maybe we can avoid sending them back to the caller by simply returning `null`.